lib/pkcs11_req.c

```diff
@@ -1098,25 +1101,35 @@ int pkcs11_sign_X509_REQ(pkcs11Context * p11Context, CK_VOID_PTR req, int output
     {
 
 	ECDSA_SIG *ecsig = ECDSA_SIG_new();
+	BIGNUM *sig_r;
+	BIGNUM *sig_s;
 
 	int siglen;
 	unsigned char *sigder = NULL;
 
 	if(ecsig==NULL) {
 	    goto err;
 	}
+	if((sig_r = BN_new()) == NULL) {
+	    goto err;
+	}
+	if((sig_s = BN_new()) == NULL) {
+      goto err;
+  }
 
-	if( BN_bin2bn( &outbuf[0], outlen/2, ecsig->r) == NULL) {
+	if( BN_bin2bn( &outbuf[0], outlen/2, sig_r) == NULL) {
 	    P_ERR();
 	    goto err;
 	}
 
-	if( BN_bin2bn( &outbuf[outlen/2], outlen/2, ecsig->s) == NULL ) {
+	if( BN_bin2bn( &outbuf[outlen/2], outlen/2, sig_s) == NULL ) {
 	    P_ERR();
 	    goto err;
 	}
+	ECDSA_SIG_set0(ecsig, sig_r, sig_s);
```

Comment:
`sig_r`and `sig_s` must be set to `NULL` after this call, to avoid double free condition (once cleanup code for these variables is added after the `err:`label, see previous comment)